runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/UnboundedReadFromBoundedSource.java

```diff
@@ -166,14 +167,18 @@ public Reader createReader(PipelineOptions options, Checkpoint<T> checkpoint)
     }
 
     @Override
-    public Coder<T> getDefaultOutputCoder() {
+    public Coder<T> getDefaultOutputCoder() throws CannotProvideCoderException {
       return boundedSource.getDefaultOutputCoder();
     }
 
     @SuppressWarnings({"rawtypes", "unchecked"})
     @Override
     public Coder<Checkpoint<T>> getCheckpointMarkCoder() {
-      return new CheckpointCoder<>(boundedSource.getDefaultOutputCoder());
+      try {
```

Comment:
It looks as an easy and analogous fix - I'll post a proposal in a separate commit for this too (in case I missed something it will be easier to revert).